google/cloud/aiplatform/explain/lit.py

```diff
@@ -75,8 +76,103 @@ def spec(self):
         return dict(self._column_types)
 
 
-class _VertexLitModel(lit_model.Model):
-    """LIT model class for the Vertex LIT integration.
+class _EndpointLitModel(lit_model.Model):
+    """LIT model class for the Vertex LIT integration with a model deployed to an endpoint.
+
+    This is used in the create_lit_model function.
+    """
+
+    def __init__(
+        self,
+        model: str,
+        input_types: "OrderedDict[str, lit_types.LitType]",  # noqa: F821
+        output_types: "OrderedDict[str, lit_types.LitType]",  # noqa: F821
+        model_id: Optional[str] = None,
+    ):
+        """Construct a VertexLitModel.
+            Args:
+              model:
+                Required. The name of the Endpoint resource. Format:
+                ``projects/{project}/locations/{location}/endpoints/{endpoint}``
+              input_types:
+                Required. An OrderedDict of string names matching the features of the model
+                as the key, and the associated LitType of the feature.
+              output_types:
+                Required. An OrderedDict of string names matching the labels of the model
+                as the key, and the associated LitType of the label.
+              model_id:
+                Optional. A string of the specific model in the endpoint to create the
+                LIT model from. If this is not set, any usable model in the endpoint is
+                used to create the LIT model.
+            Raises:
+                ValueError if the model_id was not found in the endpoint.
+        """
+        self._endpoint = aiplatform.Endpoint(model)
```

Comment:
This will only use the default environment creds or creds set in `aiplatform.init`. Accepting an `Endpoint` object will provide greater flexibility by allowing them to set creds at construction time. Is there a reason we wouldn't want to do that?

Reply:
We though that having one function that can handle different situations might have been better design, as it lowered the overhead from the user. However, I am now refactoring into another function.